src/asio_client_session_tcp_impl.cc

```diff
@@ -40,7 +40,11 @@ session_tcp_impl::session_tcp_impl(
     const std::string &host, const std::string &service,
     const boost::posix_time::time_duration &connect_timeout)
     : session_impl(io_service, connect_timeout),
-      socket_(io_service, local_endpoint) {}
+      socket_(io_service, tcp::v4()) {
```

Comment:
Is tcp::v4() required?

Reply:
Hi Tatsuhiro

Unfortunately if we just call socket(io_service) the socket will be created in the closed state.
The socket needs to be in the open state to be able to set the option.
So it's a chicken and egg problem. We want to open on local_endpoint, but need to specify reuse address be fore being able to do so.

Maybe there is an alternative to opening the socket that I'm unaware of.

Best
Alexandros

Reply:
Got it.  Thank you.